de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -533,20 +589,19 @@ public void addResourcesToSession(
         }
 
         List<IProject> projectsToShare = new ArrayList<IProject>();
-
-        for (Entry<IProject, List<IResource>> mapEntry : projectResourcesMapping
-            .entrySet()) {
-
+        for (Entry<IProject, List<IResource>> mapEntry : nextProjectNegotiation
```

Comment:
A little comment on the locking logic would not hurt. I have to check the get-function to figure out that this loop is locking. Same applies for the isEmpty call. You could make it a bit more obvious what is going on, so it is easier to check, if any of this could trigger a deadlock.

Reply:
I moved it to a separate variable, hope that helps.

In general I'm not sure what you think I should comment, for example exisiting calls like `currentProjectNegotiations.add(negotiation);` are blocking as well without further comment.